lib/rules/max-len.js

```diff
@@ -183,6 +183,23 @@ module.exports = {
                 (end.line > lineNumber || (end.line === lineNumber && end.column === line.length));
         }
 
+        /**
+         * Check if a comment is in the single line JSXEmptyExpression.
+         * @param {ASTNode} comment The comment to check.
+         * @returns {boolean} True if the comment is in the single line JSXEmptyExpression.
+         */
+        function isCommentInSingleLineJSXEmptyExpression(comment) {
+            const node = sourceCode.getNodeByRangeIndex(comment.range[0]);
+
+            if (!node || !node.parent || node.type !== "JSXEmptyExpression" || node.parent.type !== "JSXExpressionContainer") {
+                return false;
+            }
+
+            const parent = node.parent;
+
+            return parent.loc.start.line === parent.loc.end.line;
+        }
+
```

Comment:
I'm not sure what would be correct behavior when there are two comments inside, like `{ /* 1 */ /* 2 */ }`.

A small difference is that this is now a `valid` comment-only line:

```js
{
    code: "var jsx = (<>\n" +
         "  {/* this line has > 20 characters */ /* but < 100 */}\n" +
         "</>)",
    options: [{ code: 20, comments: 100 }],
    parserOptions: { ecmaFeatures: { jsx: true } }
}
```

while this is `invalid`, because a line with two comments is treated as a 'code' line:

```js
{
    code: "var foo = \n" +
        "  /* this line has > 20 characters */ /* but < 100 */\n" +
        "5;",
     options: [{ code: 20, comments: 100 }],
     errors: 1 // "This line has a length of 53. Maximum allowed is 20."
}
```

Thoughts? Maybe this doesn't matter as much.

Reply:
Although it needs to be evaluated by eslint members, In my opinion, the second case looks like a bug. :)

Reply:
I agree and would vote to do with JSX comments whatever seems to be easier for the other problem (with trailing comments).